### PR TITLE
[Feature-#4350][Alert]support feishu plugin

### LINKED ISSUE #4350: [Feature][Alert]support access to feishu
The current version (ver 1.3.2) only supports sending alarms via email and WeChat (for example: task failure, master or worker disconnection, etc.), we need to add an alarm for accessing Feishu

### PR DESCRIPTION
**Pull request by felix-thinkingdata**
Realized the feishu  robot to send warning message

this closes #4350

Test the screenshot：
image

image

### CODE REVIEW

**Comment by chengshiwen** [comment]
@felix-thinkingdata Please rename the subject (ref: [https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html](https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html)) and pass the all checks at first

File: dolphinscheduler-alert-plugin/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/FeiShuParamsConstants.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+package org.apache.dolphinscheduler.plugin.alert.feishu;
+
+public class FeiShuParamsConstants {
+
+    private FeiShuParamsConstants() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    static final String FEI_SHU_WEB_HOOK = "feishu.webhook";
```
Comment (CalvinKirs) [inline dolphinscheduler-alert-plugin/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/FeiShuParamsConstants.java:26]: I think it’s better to name it webHook, try not to carry too much contextual information

Comment (felix-thinkingdata) [reply dolphinscheduler-alert-plugin/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/FeiShuParamsConstants.java:26]: static final String WEB_HOOK = "webhook";   
 
 Is that what you mean？

Comment (CalvinKirs) [reply dolphinscheduler-alert-plugin/dolphinscheduler-alert-feishu/src/main/java/org/apache/dolphinscheduler/plugin/alert/feishu/FeiShuParamsConstants.java:26]: Right!

**Comment by felix-thinkingdata** [comment]
> @felix-thinkingdata Please rename the pr title (ref: https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html) and pass the all checks at first

After work to catch the subway, did not have time to follow the rules
下班坐地铁，没来得及弄

**Comment by felix-thinkingdata** [comment]
Sonarcloud reports the problem, I will optimize it tomorrow

**Comment by felix-thinkingdata** [comment]
> SonarCloud Quality Gate failed.
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=4594&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=4594&metric=new_coverage&view=list) [57.4% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=4594&metric=new_coverage&view=list)
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=4594&metric=new_duplicated_lines_density&view=list) [8.4% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=4594&metric=new_duplicated_lines_density&view=list)

Sonarcloud reports the problem, I will optimize it tomorrow

**Review by CalvinKirs — COMMENTED** [review]
You should add the flying book plugin dolphinscheduler-dist/src/main/provisio/dolphinscheduler.xml here

**Comment by felix-thinkingdata** [comment]
> You should add the flying book plugin dolphinscheduler-dist/src/main/provisio/dolphinscheduler.xml here

ok

**Review by CalvinKirs — COMMENTED** [review]
looks good to me
It is recommended to remove all "feishu" in the field name, do not include too much context information

**Comment by felix-thinkingdata** [comment]
> looks good to me
> It is recommended to remove all "feishu" in the field name, do not include too much context information

Done

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**